### ISSUE TITLE
scalaVersion mediator in 0.13.13 is broken for dotty-compiled projects

### ISSUE THREAD

**Issue by smarter** [sbt/sbt#2896]
We now publish a dummy `ch.epfl.lamp % scala-library % 0.1-SNAPSHOT` that depends on `dotty-library` so that users do not need to depend on the `scala-library` by hand: https://github.com/smarter/dotty-example-project/commit/3012522ae15e4711bfa171705256696b41467da1

This works perfectly fine with sbt 0.13.11 but breaks in 0.13.13 because `dotty-library` depends on `org.scala-lang % scala-library % 2.11.5`, if you take https://github.com/smarter/dotty-example-project and try to compile it with `0.13.13` you get:
```scala
[debug] dependency descriptor has been mediated: dependency: org.scala-lang#scala-library;2.11.5 {compile=[default(compile)]} => dependency: ch.epfl.lamp#scala-library;0.1-SNAPSHOT {compile=[default(compile)]}
[error] a module is not authorized to depend on itself: ch.epfl.lamp#scala-library;0.1-SNAPSHOT
```

This can be worked around by adding `ivyScala ~= (_ map (_ copy (overrideScalaVersion = false)))` to the user build, but that defeats the point of avoiding boilerplate.
Is there anything we can do in our published `scala-library` to make sure sbt does not try to rewrite its dependencies? Alternatively, could we add some special cases to sbt so that it treats `scala-library` specially? I'll note that scala/scala also needs to disable the overriding when they're bootstrapping, so this isn't a problem specific to dotty, although it's much more annoying in dotty since it affects users' build.sbt: https://github.com/scala/scala/blob/2.12.x/build.sbt#L122

**Comment by smarter** [comment]
I think this is the root cause of https://github.com/sbt/sbt/issues/2867

**Comment by smarter** [comment]
I see there's already some cases where the mediator is disabled, maybe that could be extended? https://github.com/sbt/sbt/pull/2827

**Comment by eed3si9n** [comment]
Fixed by #2897 and https://github.com/sbt/librarymanagement/pull/79
